src/RNCWebViewNativeComponent.ts

```diff
@@ -0,0 +1,274 @@
+  fraudulentWebsiteWarningEnabled?: boolean;
+  // !iOS only
+
+  allowFileAccessFromFileURLs?: boolean;
+  allowUniversalAccessFromFileURLs?: boolean;
+  applicationNameForUserAgent?: string;
+  basicAuthCredential?: Readonly<{
+    username: string;
+    password: string;
+  }>;
+  cacheEnabled?: boolean;
+  incognito?: boolean;
+  injectedJavaScript?: string;
+  injectedJavaScriptBeforeContentLoaded?: string;
+  injectedJavaScriptForMainFrameOnly?: boolean;
+  injectedJavaScriptBeforeContentLoadedForMainFrameOnly?: boolean;
+  javaScriptCanOpenWindowsAutomatically?: boolean;
+  javaScriptEnabled?: boolean;
+  webviewDebuggingEnabled?: boolean;
+  mediaPlaybackRequiresUserAction?: boolean;
+  messagingEnabled: boolean;
+  onLoadingError: DirectEventHandler<WebViewErrorEvent>;
+  onLoadingFinish: DirectEventHandler<WebViewNavigationEvent>;
+  onLoadingProgress: DirectEventHandler<WebViewNativeProgressEvent>;
+  onLoadingStart: DirectEventHandler<WebViewNavigationEvent>;
+  onHttpError: DirectEventHandler<WebViewHttpErrorEvent>;
+  onMessage: DirectEventHandler<WebViewMessageEvent>;
+  onScroll?: DirectEventHandler<ScrollEvent>;
+  onShouldStartLoadWithRequest: DirectEventHandler<ShouldStartLoadRequestEvent>;
+  showsHorizontalScrollIndicator?: boolean;
+  showsVerticalScrollIndicator?: boolean;
+  newSource: Readonly<{
+    uri?: string
+    method?: string;
+    body?: string;
+    // eslint-disable-next-line @typescript-eslint/array-type
+    headers?: ReadonlyArray<Readonly<{name: string, value: string}>>;
+    html?: string;
+    baseUrl?: string;
+  }>;
+  userAgent?: string;
+}
+
+export interface NativeCommands {
+  goBack: (viewRef: React.ElementRef<HostComponent<NativeProps>> ) => void;
+  goForward: (viewRef: React.ElementRef<HostComponent<NativeProps>>) => void;
+  reload: (viewRef: React.ElementRef<HostComponent<NativeProps>>) => void;
+  stopLoading: (viewRef: React.ElementRef<HostComponent<NativeProps>>) => void;
+  injectJavaScript: (viewRef: React.ElementRef<HostComponent<NativeProps>>, javascript: string) => void;
+  requestFocus: (viewRef: React.ElementRef<HostComponent<NativeProps>>) => void;
+  postMessage: (viewRef: React.ElementRef<HostComponent<NativeProps>>, data: string) => void;
+  // Android Only
+  loadUrl: (viewRef: React.ElementRef<HostComponent<NativeProps>>, url: string) => void;
+  clearFormData: (viewRef: React.ElementRef<HostComponent<NativeProps>>) => void;
+  clearCache: (viewRef: React.ElementRef<HostComponent<NativeProps>>, includeDiskFiles: boolean) => void;
+  clearHistory: (viewRef: React.ElementRef<HostComponent<NativeProps>>) => void;
+  // !Android Only
+}
+
+export const Commands = codegenNativeCommands<NativeCommands>({
+  supportedCommands: ['goBack', 'goForward', 'reload', 'stopLoading', 'injectJavaScript', 'requestFocus', 'postMessage', 'loadUrl', 'clearFormData', 'clearCache', 'clearHistory'],
+});
+
+export default codegenNativeComponent<NativeProps>(
+  'RNCWebView'
```

Comment:
On iOS, the parent component is RNCWebView, and its child is WKWebView. On Android, it used to be just one RNCWebView which extended the native Android WebView. Now, the parent component is RNCWebViewContainer, and its child is the old RNCWebView.

@joemun , do you know if it's okay for this to say `RNCWebView` even though the native component is `RNCWebViewContainer` on Android:

Reply:
Thanks!

> This value needs to match the name specified in the view manager (RNCWebViewModuleImpl.NAME)

To clarify, can the name say `RNCWebView` on Android in the view manager even if the native component that maps 1:1 to the react component is actually named `RNCWebViewContainer`? Also, I'm assuming that on Android and iOS, the name (in the view manager on Android and the exported name from RCT_EXPORT_MODULE in RNCWebViewManager.mm) have to be the same. Is that correct?

Reply:
On the TSX side, it's referencing the `RNCWebView` native component, which returns an instance of the `RNCWebViewContainer`. So I think this is fine.  The name here is supposed to align with the view manager.

https://github.com/discord/react-native-webview/pull/27/files#diff-8576be466ba1cccbe7c11ba839c58655910cfcea677ad1de381d2a5194089ad8R16

Reply:
> Also, I'm assuming that on Android and iOS, the name (in the view manager on Android and the exported name from RCT_EXPORT_MODULE in RNCWebViewManager.mm) have to be the same. Is that correct?

Yep